app/src/main/java/com/github/fribourgsdp/radio/data/view/UserProfileActivity.kt

```diff
@@ -170,4 +164,101 @@ open class UserProfileActivity : MyAppCompatActivity() {
                 .build()
         }
     }
+
+    override fun onPick(choice: MergeDismissImportPlaylistDialog.Choice) {
+        when (choice){
+            MergeDismissImportPlaylistDialog.Choice.MERGE -> mergePlaylist()
+            MergeDismissImportPlaylistDialog.Choice.IMPORT -> importPlaylist()
+            MergeDismissImportPlaylistDialog.Choice.DISMISS_ONLINE -> dismissOnlinePlaylist()
+        }.addOnSuccessListener {
+            user.save(this)
+        }
+
+    }
+
+    private fun mergePlaylist() : Task<Unit>{
+        Log.w(ContentValues.TAG, "MERGE")
+        return db.getUser(user.id).continueWith {
```

Comment:
What is the merging policy if two playlists happen to be have the same name ?

Reply:
It does the policy implemented in the User class. As the playlists are stored in a set and two playlists are equals given the name, it does what set does with equals object. Whether they keep the old one or put the new one, I have to check

Reply:
I have tested and it keeps the most recently added playlist. In our case, it corresponds to the playlist that comes from online. I have added a test in UserTest to demonstrate this behaviour.

Reply:
Nice ! Better be sure of what happens underneath.